forge-gui/res/cardsfolder/upcoming/nine-fingers_keene.txt

```diff
@@ -0,0 +1,17 @@
+Name:Nine-Fingers Keene
+ManaCost:1 B G U
+Types:Legendary Creature Human Rogue
+PT:4/4
+K:Menace
+K:Ward:PayLife<9>
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | Execute$ TrigPeek | CombatDamage$ True | TriggerDescription$ Whenever CARDNAME deals combat damage to a player, look at the top nine cards of your library. You may put a Gate card from among them onto the battlefield. Then if you control nine or more Gates, put the rest into your hand. Otherwise, put the rest on the bottom of your library in a random order.
+SVar:TrigPeek:DB$ PeekAndReveal | PeekAmount$ 9 | NoReveal$ True | RememberPeeked$ True | SubAbility$ PickOne
```

Comment:
Should be able to do this with a Dig instead of PeekAndReveal + ChooseCard

Reply:
My first pass during testing involved a Dig, but I ran into a problem with needing to declare the destination zone for the revealed cards. I couldn't declare either of library or hand without first making the nine-gates check. There might be a way to do this with Dig instead, but I haven't had success over the PeekAndReveal route. Any feedback or hints on this would be appreciated if able

Reply:
Since we are looking at the top nine cards of the library, they technically won't leave the top of the library, so set DestinationZone2 accordingly.
Dig then Branch similar to what you already have here should do the trick, I think.

Reply:
Resolving conversation, will revisit Nine-Fingers Keene on addition of broader Dig param support